### ISSUE TITLE
Some question about sidetree protocal

### ISSUE THREAD

**Issue by SpencerChou** [decentralized-identity/sidetree#1191]
[here is the some Questions I asked on ION repo](https://github.com/decentralized-identity/ion/issues/274)

Here are still some questions.

1) About the payload mentioned above in ION-tools, the payload can be anything I want to sign, so it can be a VC,right?

2)As I learned from the DID definition, I find that the VC/CA are handout by other organizations. The sidetree system only provide DID-publickey to verify the VC content.  So what's the difference between traditional PKI and sidetree/DPKI. The org mentioned above is the CA node, and the CA node is still under the risk of being attack. 

3) every sidetree node monitor the change on the blockchain, and then finding the source sidetree node and fetch operations through ipfs. So every mongoDB attached with one sidetree node will contain all DIDs(the mongoDB storage and  sidetree node is one-one relationship ?). is that correct?

**Comment by thehenrytsai** [comment]
1. A VC is by definition signed. Not sure if you meant you want to signed it again using `ion-tools` or you want to create a VC using ION tools.
2. A significant difference is decentralization, you don't need 3rd party authority to establish trust like traditional PKI.

>  and then finding the source sidetree node and fetch operations through ipfs.

3. ION nodes uses IPFS to fetch data, the intend is that there will be many copies of the data over time, though its true that the data would originally come from one IPFS node.

> So every mongoDB attached with one sidetree node will contain all DIDs

3. Yes.

> the mongoDB storage and sidetree node is one-one relationship ?

3. Not 100% sure what this mean, but the MongoDB instance should be used by a single ION Node.

**Comment by SpencerChou** [comment]
The institute who give out the DIDs is decentralized. But the CA organization is still centralized, right

**Comment by 2byrds** [comment]
> The institute who give out the DIDs is decentralized. But the CA organization is still centralized, right

Sounds like you are asking about the self-sovereign identity concept of a Trust Registry.  Trust Registries tell us that an issuer is 'allowed' to issue a particular VC.  For instance, if we ask the question "Is that clinic that issued a vaccination VC allowed to issue such a VC?"  You could imagine many such registries interacting between governing bodies so that we could globally determine if a vaccination VC issuer is trust worthy.  There will be a variety of centralization and decentralization depending on the use-case/environment/country/community.  Does that touch on your question?

**Comment by SpencerChou** [comment]
thanks a lot. It seems that I mess around the sidetree node with the concept of Trust Registry.

Because when I first learn the concept of DID, It says the traditional Centralized PKI is under the risk of being attack. there is a security problem with it. So I thought the DID theory must solve this problem. Now I see the IDs are decentralized. But the CA seems still under the risk of being attack, which seems the DID theory do not solve the problem.

So I am so confused

**Comment by thehenrytsai** [comment]
> It says the traditional Centralized PKI is under the risk of being attack

@SpencerChou, yes, since Centralized PKI is "centralized", it is "more possible" to be attacked and brought down in theory resulting in unavailable resolution service. Where as it should be near impossible to bring down a truly decentralized PKI system because anyone can just spin up their own node and perform their own resolutions.

Note this makes no assertion about the security risks of VC issuing entities/authorities as that's orthogonal to sidetree.

**Comment by 2byrds** [comment]
And if you want more details, minute 4 to minute 10 in this video for more details.  https://www.youtube.com/watch?v=izNZ20XSXR0
It mentions the DNS attack in 2019(?) highlighted how traditional PKI networks are vulnerable to attacks. 
 Tech like DIDs, VCs, enable the possibility of decentralization for the issuers/holders/verifiers using cryptographic techniques.  
Decentralized networks (like blockchains, but not just blockchains) can use consensus/witness mechanisms like proof of work, proof of stake, etc. to potentially decentralize to global scale and reduce your risk of single points of control/failure for the whole network.
There are still risks at all the points of centralization (any node), however the concept of rooting your trust in cryptographic methods that are more and more decentralized leads to a much harder system to attack.

**Comment by SpencerChou** [comment]
Hi there, I got new question.
I don't understand what kind of role of ipfs plays in the system. In other words, what kind of content does the ipfs module stores. How is the single sidetree node get the operations from other nodes using ipfs?What's the usage of the ipfs.

**Comment by 2byrds** [comment]
In order for sidetree implementations to store the files/hierarchy of data it needs a distributed file system.  It is only storing the CID of sidetree transactions on the blockchain. As @csuwildcat says here, the ION implementation 'is a lot IPFS and a little bit of Bitcoin' start at 8min here https://youtu.be/3UbUzA0lW9w

**Comment by SpencerChou** [comment]
> > It says the traditional Centralized PKI is under the risk of being attack
> 
> @SpencerChou, yes, since Centralized PKI is "centralized", it is "more possible" to be attacked and brought down in theory resulting in unavailable resolution service. Where as it should be near impossible to bring down a truly decentralized PKI system because anyone can just spin up their own node and perform their own resolutions.
> 
> Note this makes no assertion about the security risks of VC issuing entities/authorities as that's orthogonal to sidetree.


I kind of  understand what's going on about the VC.
Since I noticed that the DID can help improve privacy security, like birth, phone number. 
Can someone help me understand how the DID is doing this.
Cause in traditional Centralized PKI system, the CA holds the Credential which contains the privacy, they hold your publickey, you hold your own privatekey. 
In DPKI, the only difference is that the publickey is upload to blockchain which is decentralized. But I don't see how this show a higher privacy level. Cause the CA root is still vulnerable.
